src/services/fcmToken/sendChatNotificationByChannelMembers.js

```diff
@@ -0,0 +1,15 @@
+const sendChatNotification = require('./sendChatNotification');
+
+const sendChatNotificationByChannelMembers = async (
+  channelMembers = [],
+  payload = {},
+  options = {}
+) => {
+  const deduplicatedChannelMembers = [...new Set(channelMembers)];
+
+  deduplicatedChannelMembers.forEach((member) => {
+    sendChatNotification(member, payload, options);
+  });
+};
+
+module.exports = sendChatNotificationByChannelMembers;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `sendChatNotificationByChannelMembers` is using `forEach` to send notifications to each member. This could lead to performance issues if the number of members is large, as it doesn't wait for the previous notification to finish before sending the next one. 

Consider using `Promise.all` with `map` to send all notifications concurrently and wait for all of them to finish. Also, you should handle potential errors in the `sendChatNotification` function call.

```diff
-  deduplicatedChannelMembers.forEach((member) => {
-    sendChatNotification(member, payload, options);
-  });
+  await Promise.all(deduplicatedChannelMembers.map(async (member) => {
+    try {
+      await sendChatNotification(member, payload, options);
+    } catch (error) {
+      console.error(`Failed to send notification to member ${member}:`, error);
+    }
+  }));
```
This change will ensure that all notifications are sent concurrently, improving performance, and any errors that occur during the process are logged and handled appropriately.


<!-- This is an auto-generated comment by OSS CodeRabbit -->